From 2b00608317d4e733a8ec5c2eb771c5d47abb2a3d Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Tue, 24 Mar 2015 13:58:36 +0100
Subject: [PATCH] ARM: DT: ls1021a-tqmls102xa-mbls102x: add mdio bus and
 ethernet

This will add the correct mdio bus addresses for the phys
and move the logical tbi-phy to the module part

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm/boot/dts/ls1021a-tqmls1021a-mbls102x.dts | 37 +++++++++++++++++++++++
 arch/arm/boot/dts/ls1021a-tqmls1021a.dtsi         |  7 +++++
 2 files changed, 44 insertions(+)

diff --git a/arch/arm/boot/dts/ls1021a-tqmls1021a-mbls102x.dts b/arch/arm/boot/dts/ls1021a-tqmls1021a-mbls102x.dts
index d3fa676..ca54281 100644
--- a/arch/arm/boot/dts/ls1021a-tqmls1021a-mbls102x.dts
+++ b/arch/arm/boot/dts/ls1021a-tqmls1021a-mbls102x.dts
@@ -13,6 +13,12 @@
 / {
 	model = "TQMLS1021A SOM on MBLS102x";
 
+	aliases {
+		enet0_rgmii_phy = &rgmii_phy0d;
+		enet1_sgmii_phy = &sgmii_phy13;
+		enet2_rgmii_phy = &rgmii_phy05;
+	};
+
 	regulators {
 		compatible = "simple-bus";
 		#address-cells = <1>;
@@ -63,6 +69,25 @@
 	status = "okay";
 };
 
+&enet0 {
+	phy-handle = <&rgmii_phy0d>;
+	phy-connection-type = "rgmii-id";
+	status = "okay";
+};
+
+&enet1 {
+	tbi-handle = <&tbi1>;
+	phy-handle = <&sgmii_phy13>;
+	phy-connection-type = "sgmii";
+	status = "okay";
+};
+
+&enet2 {
+	phy-handle = <&rgmii_phy05>;
+	phy-connection-type = "rgmii-id";
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
 	pca9554_0: pca954@20 {
@@ -134,6 +159,18 @@
 	status = "okay";
 };
 
+&mdio0 {
+	rgmii_phy05: ethernet-phy@05 {
+		reg = <0x05>;
+	};
+	rgmii_phy0d: ethernet-phy@0d {
+		reg = <0x0d>;
+	};
+	sgmii_phy13: ethernet-phy@13 {
+		reg = <0x13>;
+	};
+};
+
 &uqe {
 	tdma: ucc@2000 {
 		compatible = "fsl,ucc-tdm";
diff --git a/arch/arm/boot/dts/ls1021a-tqmls1021a.dtsi b/arch/arm/boot/dts/ls1021a-tqmls1021a.dtsi
index e41d886..dd81b8e 100644
--- a/arch/arm/boot/dts/ls1021a-tqmls1021a.dtsi
+++ b/arch/arm/boot/dts/ls1021a-tqmls1021a.dtsi
@@ -95,6 +95,13 @@
 	status = "okay";
 };
 
+&mdio0 {
+	tbi1: tbi-phy@1f {
+		reg = <0x1f>;
+		device_type = "tbi-phy";
+	};
+};
+
 &uqe {
 	tdma: ucc@2000 {
 		compatible = "fsl,ucc-tdm";
